src/lib/api/UserAPI.ts

```diff
@@ -0,0 +1,68 @@
+import { User } from "lib/model/User";
+
+/**
+ * APIの実行結果が代入される型
+ */
+export type UserAPIResult = {
+  /**
+   * APIが返したレスポンスコード。
+   */
+  code: number;
+  /**
+   * APIが返したUser。
+   */
+  received: User[];
+};
+
+// TODO: APIとの接続が実装され次第消す
+const sampleUser = [
+  {
+    uuid: "1",
+    bio: "unlimited developer",
+    birthDay: new Date(1872, 5, 1),
+    favoriteLang: "Typescript",
+    showId: "unlimit-highchi",
+    name: "H.ichiyo",
+    pictureUrl: `${process.env.PUBLIC_URL}/higuchi.png`,
+  },
+  {
+    uuid: "2",
+    bio: "another unlimited developer",
+    birthDay: new Date(1872, 5, 1),
+    favoriteLang: "Typescript",
+    showId: "dev-ichiyo",
+    name: "Ichiyo Higuchi",
+    pictureUrl: `${process.env.PUBLIC_URL}/higuchi.png`,
+  },
+];
+
+/**
+ * Userに関するAPIを叩くクラス。
+ */
+export class UserAPI {
+  /**
+   * 名前からユーザーを検索する。
+   */
+  static async searchUserFromName(query: string): Promise<UserAPIResult> {
```

Comment:
```suggestion
export namespace UserAPI {
  /**
   * 名前からユーザーを検索する。
   */
  export async function searchUserFromName(query: string): Promise<UserAPIResult> {
```
この目的なら `namespace` のほうがいいかも
(`TagAPI` など他の API 系も同様)